front/src/modules/simulationResult/components/SpaceTimeChartWrapper/SpaceTimeChartWrapper.tsx

```diff
@@ -227,70 +287,75 @@ const SpaceTimeChartWrapper = ({
           zones,
           tracks,
           loading,
-        }) => ({
-          id: operationalPointId,
-          position: operationalPointPosition,
-          size: (tracks?.length || 0) * TRACK_HEIGHT_CONTAINER + DEFAULT_THEME.timeCaptionsSize,
-          spaceTimeChartNode: (
-            <TrackOccupancyCanvas
-              position={operationalPointPosition}
-              tracks={tracks || []}
-              occupancyZones={(zones || []).map((zone) => {
-                const path = pathsIndex[zone.trainId];
-                if (!path) return zone;
-                const pathStyle = getPathStyle(
-                  hoveredItem,
-                  path,
-                  !!draggingState,
-                  subCategories,
-                  timetableItemsWithDetails,
-                  selectedTrainId
-                );
-                return {
-                  ...zone,
-                  color: pathStyle.color,
-                  size: pathStyle.level === 1 ? 2 : undefined,
-                };
-              })}
-              selectedTrainId={selectedTrainId}
-              onClose={() => onCloseOccupancyLayer?.(waypointId)}
-              topPadding={BASE_WAYPOINT_HEIGHT}
-            />
-          ),
-          manchetteNode: (
-            <TrackOccupancyManchette tracks={tracks || []}>
-              <div className="waypoint-wrapper flex justify-start">
-                <WaypointComponent
-                  waypoint={{
-                    id: waypointId,
-                    name: (
-                      <div className="d-flex flex-row align-items-center">
-                        {operationalPointName || operationalPointId}
-                        {loading && (
-                          <Spinner className="ml-2 small" spinnerClassName="spinner-border-sm" />
-                        )}
-                      </div>
-                    ),
-                    position: operationalPointPosition,
-                    onClick: handleWaypointClick,
-                  }}
-                  waypointRef={activeWaypointRef}
-                  isActive={false}
-                  isMenuActive={false}
-                />
-              </div>
-            </TrackOccupancyManchette>
-          ),
-        })
-      ) || []
-    );
-  }, [
-    trackOccupancyDiagramsData,
-    activeWaypointId,
-    timetableItemsWithDetails,
-    selectedTrainId,
-    hoveredItem,
-  ]);
+        }) => {
+          const baseZones = baseZonesByWaypointId.get(waypointId) ?? zones ?? [];
```

Comment:
Why do you need `baseZonesByWaypointId.get(waypointId)` ? I think zones is already the `zones` of the waypoint
```suggestion
          const baseZones = zones ?? [];
```